File: TelerikUIExamples/TelerikUIExamples/DefaultAnimation.m
//
//  DefaultAnimation.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "DefaultAnimation.h"
#import <TelerikUI/TelerikUI.h>

@implementation DefaultAnimation
{
    TKChart *_chart;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    
    if(self) {
        
        NSArray *names = @[ @"Area Series", @"Pie Series", @"Line Series", @"Scatter Series", @"Bar Series", @"Column Series" ];
        UIUserInterfaceIdiom idiom = [[UIDevice currentDevice] userInterfaceIdiom];

        [self addOption:[self nameForOption:0 names:names idiom:idiom] selector:@selector(setupAreaSeries)];
        [self addOption:[self nameForOption:1 names:names idiom:idiom] selector:@selector(setupPieSeries)];
        [self addOption:[self nameForOption:2 names:names idiom:idiom] selector:@selector(setupLineSeries)];
        [self addOption:[self nameForOption:3 names:names idiom:idiom] selector:@selector(setupScatterSeries)];
        [self addOption:[self nameForOption:4 names:names idiom:idiom] selector:@selector(setupBarSeries)];
        [self addOption:[self nameForOption:5 names:names idiom:idiom] selector:@selector(setupColumnSeries)];
    }
    
    return self;
}

- (NSString*)nameForOption:(NSInteger)index names:(NSArray*)names idiom:(UIUserInterfaceIdiom)idiom
{
    NSString *name = names[index];
    if (idiom != UIUserInterfaceIdiomPad) {
        name = [NSString stringWithFormat:@"%@ Animation", name];
    }
    return name;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	
    _chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    _chart.allowAnimations = YES;
    [self.view addSubview:_chart];
    
    [self setupAreaSeries];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)setupLineSeries
{
    [_chart removeAllData];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    
    for (int i = 0; i< 10; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:@(i) Y:@(arc4random() % 100)]];
    }
    
    TKChartLineSeries *lineSeries = [[TKChartLineSeries alloc] initWithItems:array];
    lineSeries.selectionMode = TKChartSeriesSelectionModeSeries;
    [_chart addSeries: lineSeries];
}

- (void)setupAreaSeries
{
    [_chart removeAllData];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    
    for (int i = 0; i< 10; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:@(i) Y:@(arc4random() % 100)]];
    }
    
    TKChartAreaSeries *areaSeries = [[TKChartAreaSeries alloc] initWithItems:array];
    areaSeries.selectionMode = TKChartSeriesSelectionModeSeries;
    [_chart addSeries: areaSeries];
}

- (void)setupPieSeries
{
    [_chart removeAllData];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    [array addObject:[[TKChartDataPoint alloc] initWithName:@"Google" value:@20]];
    [array addObject:[[TKChartDataPoint alloc] initWithName:@"Apple" value:@30]];
    [array addObject:[[TKChartDataPoint alloc] initWithName:@"Microsoft" value:@10]];
    [array addObject:[[TKChartDataPoint alloc] initWithName:@"IBM" value:@5]];
    [array addObject:[[TKChartDataPoint alloc] initWithName:@"Oracle" value:@8]];
    TKChartPieSeries *series = [[TKChartPieSeries alloc] initWithItems:array];
    series.selectionMode = TKChartSeriesSelectionModeDataPoint;
    [_chart addSeries:series];
}

- (void)setupScatterSeries
{
    [_chart removeAllData];
    
    NSMutableArray *points = [[NSMutableArray alloc] init];
    for (int i = 0; i < 100; i++) {
        [points addObject:[[TKChartDataPoint alloc] initWithX:@(arc4random() % 1450) Y:@(arc4random()%150)]];
    }
    
    TKChartScatterSeries *scatterSeries = [[TKChartScatterSeries alloc] initWithItems:points];
    scatterSeries.selectionMode = TKChartSeriesSelectionModeSeries;
    [_chart addSeries: scatterSeries];
}

- (void)setupBarSeries
{
    [_chart removeAllData];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    
    for (int i = 0; i< 10; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:@(arc4random() % 100) Y:@(i)]];
    }
    
    TKChartBarSeries *barSeries = [[TKChartBarSeries alloc] initWithItems:array];
    barSeries.selectionMode = TKChartSeriesSelectionModeSeries;
    [_chart addSeries: barSeries];
}

- (void)setupColumnSeries
{
    [_chart removeAllData];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    
    for (int i = 0; i< 10; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:@(i) Y:@(arc4random() % 100)]];
    }
    
    TKChartColumnSeries *columnSeries = [[TKChartColumnSeries alloc] initWithItems:array];
    columnSeries.selectionMode = TKChartSeriesSelectionModeSeries;
    [_chart addSeries:columnSeries];
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DefaultAnimation.swift
//
//  DefaultAnimation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation
import QuartzCore

class DefaultAnimation: ExampleViewController {
    
    let chart = TKChart()
    
    override init() {
        super.init()

        let names = ["Area Series", "Pie Series", "Line Series", "Scatter Series", "Bar Series", "Column Series"]
        let idiom = UIDevice.currentDevice().userInterfaceIdiom
        
        self.addOption(self.nameForOption(0, names: names, idiom: idiom)) { self.setupAreaSeries() }
        self.addOption(self.nameForOption(1, names: names, idiom: idiom)) { self.setupPieSeries() }
        self.addOption(self.nameForOption(2, names: names, idiom: idiom)) { self.setupLineSeries() }
        self.addOption(self.nameForOption(3, names: names, idiom: idiom)) { self.setupScatterSeries() }
        self.addOption(self.nameForOption(4, names: names, idiom: idiom)) { self.setupBarSeries() }
        self.addOption(self.nameForOption(5, names: names, idiom: idiom)) { self.setupColumnSeries() }
        
        self.selectedOption = 2
    }
    
    override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBounds
        chart.autoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight
        chart.allowAnimations = true
        self.view.addSubview(chart)
        
        self.setupLineSeries()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func nameForOption(index: NSInteger, names: NSArray, idiom: UIUserInterfaceIdiom) -> NSString {
        var name = names[index] as NSString
        if (idiom != UIUserInterfaceIdiom.Pad) {
            name = NSString(format: "%@ Animation", name)
        }
        return name
    }
    
    func setupLineSeries() {
        chart.removeAllData()
        
        let array = NSMutableArray()
        for i in 0..<10 {
            array.addObject(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let lineSeries = TKChartLineSeries(items: array)
        lineSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(lineSeries)
    }
    
    func setupAreaSeries() {
        chart.removeAllData()
        
        let array = NSMutableArray()
        for i in 0..<10 {
            array.addObject(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let areaSeries = TKChartAreaSeries(items: array)
        areaSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(areaSeries)
    }
    
    func setupPieSeries() {
        chart.removeAllData()
        
        let array = NSMutableArray()
        array.addObject(TKChartDataPoint(value: 20, name: "Google"))
        array.addObject(TKChartDataPoint(value: 30, name: "Apple"))
        array.addObject(TKChartDataPoint(value: 10, name: "Microsoft"))
        array.addObject(TKChartDataPoint(value: 5, name: "IBM"))
        array.addObject(TKChartDataPoint(value: 8, name: "Oracle"))
        
        let series = TKChartPieSeries(items: array)
        series.selectionMode = TKChartSeriesSelectionMode.DataPoint
        chart.addSeries(series)
    }
    
    func setupScatterSeries() {
        chart.removeAllData()
        
        let points = NSMutableArray()
        for i in 0..<100 {
            points.addObject(TKChartDataPoint(x: Int(arc4random() % 1450), y: Int(arc4random() % 150)))
        }
        
        let scatterSeries = TKChartScatterSeries(items: points)
        scatterSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(scatterSeries)
    }
    
    func setupBarSeries() {
        chart.removeAllData()
        
        let array = NSMutableArray()
        for i in 0..<10 {
            array.addObject(TKChartDataPoint(x: Int(arc4random() % 100), y: i))
        }
        
        let barSeries = TKChartBarSeries(items: array)
        barSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(barSeries)
    }
    
    func setupColumnSeries() {
        chart.removeAllData()
        
        let array = NSMutableArray()
        for i in 0..<10 {
            array.addObject(TKChartDataPoint(x: i, y: Int(arc4random() % 100)))
        }
        
        let columnSeries = TKChartColumnSeries(items: array)
        columnSeries.selectionMode = TKChartSeriesSelectionMode.Series
        chart.addSeries(columnSeries)
    }
}
